noUncheckedIndexedAccess not added to server tsconfig

### Environment

- Operating System: Linux
- Node Version:     v20.19.1
- Nuxt Version:     4.0.3
- CLI Version:      3.27.0
- Nitro Version:    2.12.4
- Package Manager:  pnpm@8.15.6
- Builder:          -
- User Config:      compatibilityDate, devtools
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/nuxt-starter-mp3tj1qs

### Describe the bug

In a fresh Nuxt 4 project with the new single composite `tsconfig.json`, `noUncheckedIndexedAccess` is enabled in `app/` and `shared/` but not in `server/`. This is inconvenient DX. The whole project should use the same strictness setup.

### Additional context

There were previous similar issues: #25789, #30284 (closed in favor of #22572); despite all of them closed or marked as resolved, the problem persists in Nuxt 4.0.3.

### Logs

```shell-script
❯ pnpm nuxt typecheck

 ERROR  (node:96) ExperimentalWarning: WASI is an experimental feature and might change at any time                       9:51:15 AM
(Use node --trace-warnings ... to show where the warning was created)

app/test.ts:2:13 - error TS2532: Object is possibly 'undefined'.

2 console.log(a[0] + 1);
              ~~~~

shared/test.ts:2:13 - error TS2532: Object is possibly 'undefined'.

2 console.log(a[0] + 1);
              ~~~~


Found 2 errors.
```

--- a/packages/nitro-server/src/index.ts
+++ b/packages/nitro-server/src/index.ts
@@ -222,6 +222,7 @@ export async function bundle (nuxt: Nuxt & { _nitro?: Nitro }): Promise<void> {
         compilerOptions: {
           lib: ['esnext', 'webworker', 'dom.iterable'],
           skipLibCheck: true,
+          noUncheckedIndexedAccess: true,
         },
         include: [
           join(nuxt.options.buildDir, 'types/nitro-nuxt.d.ts'),

